docs/modules/getting_started/pages/first_steps.adoc

```diff
@@ -0,0 +1,181 @@
+
+First, make sure all pods are ready:
+
+[source]
+----
+include::example$code/test-nifi.sh[tag=wait-nifi-rollout]
+----
+
+Then make sure the StatefulSets are ready:
+
+[source]
+----
+kubectl get statefulset
+----
+
+The output should show all pods ready:
+
+[source]
+----
+NAME                                 READY   AGE
+simple-nifi-node-default             2/2     5m
+simple-zk-server-default             3/3     7m
+----
+
+=== Test the NiFi cluster
+
+In order to test the NiFi cluster we use curl and jq. Since NiFi requires authentication, the log-in credentials are required to request a JWT to access to the REST API. Lets read the credentials from the deployed secret and store them into a variable:
+
+[source]
+----
+include::example$code/test-nifi.sh[tag=get-nifi-credentials]
+----
+
+Since we are accessing the cluster externally, and NiFi is a bit tricky in terms of allowing access from certain urls, a standard port-forward and access via localhost does not work. The following snippet retrieves the external node ip of one of the NiFi pods and the HTTPS port exposed in the NodePort service:
+
+[source]
+----
+include::example$code/test-nifi.sh[tag=get-nifi-host]
+----
+
+With that stored in another variable, we can access the NiFi REST API in order to retrieve a JWT, which is also stored in a variable for later access:
+
+[source]
+----
+include::example$code/test-nifi.sh[tag=get-nifi-token]
+----
+
+With the JWT ready, the NiFi REST API can be accessed properly. This is done in a loop because the NiFi cluster requires some time in order get fully ready. Leader election and synchronization may take some time. It basically checks that two nodes are successfully registered and connected:
+
+[source]
+----
+include::example$code/test-nifi.sh[tag=nifi-test-loop]
+----
+
+This script will loop some time until the NiFi cluster is fully ready. For completeness, here is the full bash script. Ignore the `#tag::` and `#end::` comments, they are used as templates for this getting started guide:
+
+[source]
+----
+include::example$code/test-nifi.sh[]
+----
+
+Run this script via:
+
+[source]
+----
+include::example$code/getting-started.sh[tag=nifi-tests]
+----
+
+Congratulations, you successfully created and tested your first NiFi cluster.
+
+=== Access NiFi the web interface
+
+You can the connection settings for the NiFi cluster via script or stackablectl.
+
+==== Via kubectl and jq
+
+In order to access the NiFi web interface you can use the following script to extract the `<ip>:<port>` settings.
+
+[source]
+----
+kubectl get svc simple-nifi -o json | jq -r --argfile endpoints <(kubectl get endpoints simple-nifi -o json) --argfile nodes <(kubectl get nodes -o json) '($nodes.items[] | select(.metadata.name == $endpoints.subsets[].addresses[].nodeName) | .status.addresses | map(select(.type == "ExternalIP" or .type == "InternalIP")) | min_by(.type) | .address | tostring) + ":" + (.spec.ports[] | select(.name == "https") | .nodePort | tostring)'
```

Comment:
This line produces the perfect output but it is hard to understand and in my opinion not suitable for a getting started guide. I would break it up into several simple steps, e.g.:

```bash
# Get node names where NiFi is running on:
$ kubectl get endpoints simple-nifi \
    --output=jsonpath='{.subsets[*].addresses[*].nodeName}'
stackable-data-platform-worker3 stackable-data-platform-worker2

# Pick a node and get its internal or external IP address (some columns omitted)
$ kubectl get node stackable-data-platform-worker3 --output=wide
NAME                                    INTERNAL-IP   EXTERNAL-IP
stackable-data-platform-worker3         172.18.0.2    <none>

# Get the service port
$ kubectl get service simple-nifi
NAME          TYPE       CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
simple-nifi   NodePort   10.96.168.210   <none>        8443:30417/TCP   5m

# Use https://172.18.0.2:30417/nifi to access the NiFi web interface
```

However, I would move all code snippets to `getting-started.sh` because there you can test the syntax easily and the code is processed by the linter.

Reply:
Removed the complex testing and just showing how to access the web url. Needed to change this up a little for automation in the getting_started.sh. (Still a little complex for the kubectl part)

Reply:
ok